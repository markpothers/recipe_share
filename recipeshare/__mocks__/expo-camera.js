

export const Camera = {
	requestCameraPermissionsAsync: () => {
		return Promise.resolve({granted: true})
	}
}
