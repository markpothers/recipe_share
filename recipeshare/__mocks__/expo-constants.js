export default {
	manifest: {
		ios: {
			buildNumber: 22
		},
		android: {
			versionCode: 22
		}
	}
}
