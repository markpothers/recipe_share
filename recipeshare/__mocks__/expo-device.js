let deviceType = 1

export const setMockDeviceType = (type) => {
    deviceType = type
}

export const getDeviceTypeAsync = jest.fn(() => {
    // console.warn('Im here!')
    return new Promise(resolve => {
        resolve(deviceType)
    })
});