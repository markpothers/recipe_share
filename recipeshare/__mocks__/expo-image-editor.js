import React, { useEffect } from "react";
import { View } from "react-native";

export const ImageEditor = (props) => {
	useEffect(()=>{
		props.onEditingComplete({
			cancelled: false,
			uri: props.imageUri,
		})
	}, [props])
	return <View testID={"mockImageEditor"}></View>;
};
