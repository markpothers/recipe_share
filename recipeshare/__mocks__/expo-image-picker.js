

export const launchCameraAsync = () => {
	return Promise.resolve({
		canceled: false,
		assets: [{uri: "mockCameraUri"}]
	})
}

export const launchImageLibraryAsync = () => {
	return Promise.resolve({
		canceled: false,
		assets: [{uri: "mockLibraryUri"}]
	})
}
