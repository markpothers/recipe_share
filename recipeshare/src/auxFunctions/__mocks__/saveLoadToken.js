export function saveToken () {
	console.log("mock save token!!!!!")
	return null
}

export function loadToken () {
	return null
}
