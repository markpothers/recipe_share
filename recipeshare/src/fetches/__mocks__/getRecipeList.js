import { mockRecipeList } from '../../../__mocks__/recipeList/mockRecipeList'

export const getRecipeList = () => {
    // console.warn('mock fetch was called')
    // console.warn(mockRecipeList)
    return new Promise((resolve, reject) => {
        resolve(mockRecipeList)
    })
}
