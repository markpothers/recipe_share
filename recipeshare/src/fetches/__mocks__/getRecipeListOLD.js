import { mockRecipeList } from "../../../__mocks__/mockRecipeList"

export const getRecipeList = () => {
    console.warn("mock fetch was called")
    // console.warn(mockRecipeList)
    return new Promise((resolve, reject) => {
        resolve(mockRecipeList)
    })
}
