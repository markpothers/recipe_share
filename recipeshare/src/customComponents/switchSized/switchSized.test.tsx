import React from "react"
import { render, fireEvent } from "@testing-library/react-native"
import { Platform } from "react-native"
import SwitchSized from "./switchSized"
import { setMockDeviceType } from "../../../__mocks__/expo-device"

describe("SwitchSized", () => {

	let mockChangeFunction

	beforeEach(async () => {
		mockChangeFunction = jest.fn()
	})

	test("can be rendered on an iPhone with reduced size switches where switch is on", async () => {
		setMockDeviceType(1) //phone
		const { toJSON } = render(
			<SwitchSized
					value={true}
					onValueChange={mockChangeFunction}
				/>
		)
		expect(toJSON()).toMatchSnapshot()
	})

	test("can be rendered on an iPhone with reduced size switches where switch is off", async () => {
		setMockDeviceType(1)
		const { toJSON } = render(
			<SwitchSized
					value={false}
					onValueChange={mockChangeFunction}
				/>
		)
		expect(toJSON()).toMatchSnapshot()
	})

	test("can be rendered on an android switch is off", async () => {
		Platform.OS = "android"
		setMockDeviceType(1)
		const { toJSON } = render(
			<SwitchSized
					value={false}
					onValueChange={mockChangeFunction}
				/>
		)
		expect(toJSON()).toMatchSnapshot()
	})

	test("can be rendered on an android where switch is on", async () => {
		Platform.OS = "android"
		setMockDeviceType(1)
		const { toJSON } = render(
			<SwitchSized
					value={true}
					onValueChange={mockChangeFunction}
				/>
		)
		expect(toJSON()).toMatchSnapshot()
	})

	test("can be rendered on iPad with regular switch where switch is on", async () => {
		setMockDeviceType(2) //iPad
		const { toJSON } = render(
			<SwitchSized
					value={true}
					onValueChange={mockChangeFunction}
				/>
		)
		expect(toJSON()).toMatchSnapshot()
	})

	test("can be rendered on iPad with regular switch where switch is off", async () => {
		setMockDeviceType(2)
		const { toJSON } = render(
			<SwitchSized
					value={false}
					onValueChange={mockChangeFunction}
				/>
		)
		expect(toJSON()).toMatchSnapshot()
	})

	test("can be rendered and switched off", async () => {
		const { getByRole } = render(
			<SwitchSized
					value={true}
					onValueChange={mockChangeFunction}
				/>
		)
		fireEvent(getByRole("switch"), "onValueChange")
		expect(mockChangeFunction).toHaveBeenCalled()
		expect(mockChangeFunction).toHaveBeenCalledTimes(1)
	})

	test("can be rendered and switched on", async () => {
		const { getByRole } = render(
			<SwitchSized
					value={false}
					onValueChange={mockChangeFunction}
				/>
		)
		fireEvent(getByRole("switch"), "onValueChange")
		expect(mockChangeFunction).toHaveBeenCalled()
		expect(mockChangeFunction).toHaveBeenCalledTimes(1)
	})

	test("can be rendered and switched on and off", async () => {
		const { getByRole } = render(
			<SwitchSized
					value={false}
					onValueChange={mockChangeFunction}
				/>
		)
		fireEvent(getByRole("switch"), "onValueChange")
		fireEvent(getByRole("switch"), "onValueChange")
		expect(mockChangeFunction).toHaveBeenCalled()
		expect(mockChangeFunction).toHaveBeenCalledTimes(2)
	})

})
