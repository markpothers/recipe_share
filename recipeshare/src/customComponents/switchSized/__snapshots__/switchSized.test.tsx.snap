// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SwitchSized can be rendered on an android switch is off 1`] = `
<AndroidSwitch
  accessibilityRole="switch"
  enabled={true}
  on={false}
  onChange={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Array [
      null,
      null,
      Object {
        "height": 46.690000000000005,
        "marginVertical": 6.67,
      },
    ]
  }
  thumbTintColor="#eaeaea"
  trackColorForFalse="#64715599"
  trackColorForTrue="#5c8a5199"
  trackTintColor="#64715599"
/>
`;

exports[`SwitchSized can be rendered on an android where switch is on 1`] = `
<AndroidSwitch
  accessibilityRole="switch"
  enabled={true}
  on={true}
  onChange={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Array [
      null,
      null,
      Object {
        "height": 46.690000000000005,
        "marginVertical": 6.67,
      },
    ]
  }
  thumbTintColor="#4b7142"
  trackColorForFalse="#64715599"
  trackColorForTrue="#5c8a5199"
  trackTintColor="#5c8a5199"
/>
`;

exports[`SwitchSized can be rendered on an iPhone with reduced size switches where switch is off 1`] = `
<RCTSwitch
  accessibilityRole="switch"
  onChange={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  onTintColor="#5c8a5199"
  style={
    Array [
      Object {
        "height": 31,
        "width": 51,
      },
      Array [
        null,
        null,
        null,
      ],
    ]
  }
  thumbTintColor={null}
  tintColor={null}
  value={false}
/>
`;

exports[`SwitchSized can be rendered on an iPhone with reduced size switches where switch is on 1`] = `
<RCTSwitch
  accessibilityRole="switch"
  onChange={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  onTintColor="#5c8a5199"
  style={
    Array [
      Object {
        "height": 31,
        "width": 51,
      },
      Array [
        null,
        null,
        null,
      ],
    ]
  }
  thumbTintColor={null}
  tintColor={null}
  value={true}
/>
`;

exports[`SwitchSized can be rendered on iPad with regular switch where switch is off 1`] = `
<AndroidSwitch
  accessibilityRole="switch"
  enabled={true}
  on={false}
  onChange={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Array [
      null,
      null,
      Object {
        "height": 46.690000000000005,
        "marginVertical": 6.67,
      },
    ]
  }
  thumbTintColor="#eaeaea"
  trackColorForFalse="#64715599"
  trackColorForTrue="#5c8a5199"
  trackTintColor="#64715599"
/>
`;

exports[`SwitchSized can be rendered on iPad with regular switch where switch is on 1`] = `
<AndroidSwitch
  accessibilityRole="switch"
  enabled={true}
  on={true}
  onChange={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Array [
      null,
      null,
      Object {
        "height": 46.690000000000005,
        "marginVertical": 6.67,
      },
    ]
  }
  thumbTintColor="#4b7142"
  trackColorForFalse="#64715599"
  trackColorForTrue="#5c8a5199"
  trackTintColor="#5c8a5199"
/>
`;
